macros/src/actix.rs

```diff
@@ -1069,6 +1061,210 @@ pub fn emit_v2_security(input: TokenStream) -> TokenStream {
     gen.into()
 }
 
+/// Actual parser and emitter for `Apiv2Header` derive macro.
+pub fn emit_v2_header(input: TokenStream) -> TokenStream {
+    let item_ast = match crate::expect_struct_or_enum(input) {
+        Ok(i) => i,
+        Err(ts) => return ts,
+    };
+
+    if let Some(empty) = check_empty_schema(&item_ast) {
+        return empty;
+    }
+
+    let name = &item_ast.ident;
+    // Add `Apiv2Schema` bound for impl if the type is generic.
+    let mut generics = item_ast.generics.clone();
+    let bound = syn::parse2::<TraitBound>(quote!(paperclip::v2::schema::Apiv2Schema))
+        .expect("expected to parse trait bound");
+    generics.type_params_mut().for_each(|param| {
+        param.bounds.push(bound.clone().into());
+    });
+
+    let opt_impl = add_optional_impl(name, &generics);
+    let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
+
+    let mut header_definitions = vec![];
+
+    let valid_attrs = vec!["description", "name", "format"];
+    let invalid_attr_msg = format!(
+        "Invalid macro attribute. Should be named attribute {:?}",
+        valid_attrs
+    );
+
+    fn quote_format(format: String) -> proc_macro2::TokenStream {
+        match &*format {
+            "int32" => quote! { Some(paperclip::v2::models::DataTypeFormat::Int32) },
+            "int64" => quote! { Some(paperclip::v2::models::DataTypeFormat::Int64) },
+            "float" => quote! { Some(paperclip::v2::models::DataTypeFormat::Float) },
+            "double" => quote! { Some(paperclip::v2::models::DataTypeFormat::Double) },
+            "byte" => quote! { Some(paperclip::v2::models::DataTypeFormat::Byte) },
+            "binary" => quote! { Some(paperclip::v2::models::DataTypeFormat::Binary) },
+            "data" => quote! { Some(paperclip::v2::models::DataTypeFormat::Date) },
+            "datetime" => quote! { Some(paperclip::v2::models::DataTypeFormat::DateTime) },
```

Comment:
this should be date-time (with the dash in the middle) ?

Reply:
I did added both (`"datetime" | "date-time"`) because serde rename it to `date-time` but the `ToString` implementation return `datetime`